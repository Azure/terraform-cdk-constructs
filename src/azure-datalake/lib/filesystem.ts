import {
  StorageDataLakeGen2Filesystem,
  StorageDataLakeGen2FilesystemConfig,
} from "@cdktf/provider-azurerm/lib/storage-data-lake-gen2-filesystem";
import { Construct } from "constructs";

export class DataLakeFilesystem extends Construct {
  public readonly name: string;
  public readonly filesystem: StorageDataLakeGen2Filesystem;

  constructor(
    scope: Construct,
    id: string,
    props: StorageDataLakeGen2FilesystemConfig,
  ) {
    super(scope, id);
    this.filesystem = new StorageDataLakeGen2Filesystem(
      this,
      "filesystem",
      props,
    );
  }
}
