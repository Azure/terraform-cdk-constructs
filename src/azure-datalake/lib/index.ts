export * from "./filesystem";
export * from "./path";
export * from "./datalake";
