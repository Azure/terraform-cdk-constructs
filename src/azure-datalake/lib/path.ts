import {
  StorageDataLakeGen2Path,
  StorageDataLakeGen2PathConfig,
  StorageDataLakeGen2PathAce,
} from "@cdktf/provider-azurerm/lib/storage-data-lake-gen2-path";
import { Construct } from "constructs";

export interface DataLakePathConfig {
  readonly group?: string;
  readonly owner?: string;
  readonly ace?: StorageDataLakeGen2PathAce[];
}

export class DataLakePath extends Construct {
  public readonly name: string;
  public readonly filesystem: StorageDataLakeGen2Path;

  constructor(
    scope: Construct,
    id: string,
    props: StorageDataLakeGen2PathConfig,
  ) {
    super(scope, id);
    this.name = id;
    this.filesystem = new StorageDataLakeGen2Path(this, "path", props);
  }
}
