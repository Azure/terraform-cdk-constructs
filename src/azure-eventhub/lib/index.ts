export * from "./authorization";
export * from "./eventhubcluster";
export * from "./consumer";
export * from "./instance";
export * from "./kusto-connection";
export * from "./namespace";
