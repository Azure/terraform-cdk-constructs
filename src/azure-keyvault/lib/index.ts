export * from './certificate'
export * from './key'
export * from './policy'
export * from './secret'
export * from './vault'

