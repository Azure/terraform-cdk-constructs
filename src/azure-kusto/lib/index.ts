export * from "./cluster";
export * from "./compute-specification";
export * from "./database";
