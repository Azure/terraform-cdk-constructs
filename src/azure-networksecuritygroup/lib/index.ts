export * from './preconfigured-rules';
export * from './securitygroup';
