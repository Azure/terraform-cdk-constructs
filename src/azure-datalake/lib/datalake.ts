import { ResourceGroup } from "@cdktf/provider-azurerm/lib/resource-group";
import { StorageDataLakeGen2FilesystemConfig } from "@cdktf/provider-azurerm/lib/storage-data-lake-gen2-filesystem";
import { Construct } from "constructs";
import { DataLakeFilesystem, DataLakeFilesystemConfig } from "./filesystem";
import * as sa from "../../azure-storageaccount";
import { AzureResource } from "../../core-azure";

export class DataLake extends AzureResource {
  readonly props: any;
  public resourceGroupName: string;
  public resourceGroup: ResourceGroup;
  public readonly filesystems: Map<string, DataLakeFilesystem>;

  public readonly storageAccount: sa.Account;

  constructor(scope: Construct, id: string, props: any) {
    super(scope, id);
    this.filesystems = new Map<string, DataLakeFilesystem>();
    // Default values
    this.props = {
      name: "test42348808",
      location: "eastus",
      ...props,
    };

    this.resourceGroup = this.setupResourceGroup(props);
    this.resourceGroupName = this.resourceGroup.name;

    this.storageAccount = new sa.Account(scope, id + "storageAccount", {
      ...this.props,
      accountTier: "Standard",
      accountKind: "StorageV2",
      isHnsEnabled: true,
      resourceGroup: this.resourceGroup,
    });
  }

  public addDataLakeFilesystem(
    name: string,
    props: DataLakeFilesystemConfig,
  ): DataLakeFilesystem {
    if (this.filesystems.has(name)) {
      throw new Error(`Filesystem '${name}' already exists.`);
    }

    var config: StorageDataLakeGen2FilesystemConfig = {
      name: name,
      storageAccountId: this.storageAccount.id,
      ...props,
    };

    const newFilesystem = new DataLakeFilesystem(this, name, config);

    this.filesystems.set(name, newFilesystem);
    return newFilesystem;
  }
}
