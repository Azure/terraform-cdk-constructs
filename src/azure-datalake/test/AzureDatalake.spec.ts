import { AzurermProvider } from "@cdktf/provider-azurerm/lib/provider";
import { Testing, TerraformStack } from "cdktf";
import { TerraformPlan } from "../../testing";

import "cdktf/lib/testing/adapters/jest";
import * as storage from "..";

describe("Azure Storage Account With Defaults", () => {
  let stack: TerraformStack;
  let fullSynthResult: any;

  beforeEach(() => {
    const app = Testing.app();
    stack = new TerraformStack(app, "test");

    new AzurermProvider(stack, "azureFeature", { features: {} });

    // Create a Storage Account with the defined rules
    new storage.DataLake(stack, "storageaccount", {
      name: "test42348808",
      location: "eastus",
    });

    fullSynthResult = Testing.fullSynth(stack); // Save the result for reuse
  });

  it("renders an Azure Storage Account with defaults and checks snapshot", () => {
    expect(Testing.synth(stack)).toMatchSnapshot(); // Compare the already prepared stack
  });

  it("check if the produced terraform configuration is valid", () => {
    expect(fullSynthResult).toBeValidTerraform(); // Use the saved result
  });

  it("check if this can be planned", () => {
    TerraformPlan(fullSynthResult); // Use the saved result
  });
});
