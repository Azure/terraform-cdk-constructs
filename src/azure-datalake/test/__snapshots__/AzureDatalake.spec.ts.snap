// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Azure Storage Account With Defaults renders an Azure Storage Account with defaults and checks snapshot 1`] = `
"{
  "provider": {
    "azurerm": [
      {
        "features": {
        }
      }
    ]
  },
  "resource": {
    "azurerm_resource_group": {
      "storageaccount_rg_F4452564": {
        "location": "eastus",
        "name": "rg-test42348808"
      }
    },
    "azurerm_storage_account": {
      "storageaccountstorageAccount_storageaccount_0DD35845": {
        "access_tier": "Hot",
        "account_kind": "StorageV2",
        "account_replication_type": "LRS",
        "account_tier": "Standard",
        "enable_https_traffic_only": true,
        "is_hns_enabled": true,
        "location": "eastus",
        "min_tls_version": "TLS1_2",
        "name": "test42348808",
        "resource_group_name": "\${azurerm_resource_group.storageaccount_rg_F4452564.name}",
        "shared_access_key_enabled": true
      }
    }
  },
  "terraform": {
    "required_providers": {
      "azurerm": {
        "source": "azurerm",
        "version": "3.70.0"
      }
    }
  }
}"
`;
